### PR TITLE
Extract repetitive routines in rich text sanitization #6211 

### LINKED ISSUE #6211: Extract repetitive routines in rich-text sanitization
The following routine is repeated in many places:
```java
text == null ? null : new Text(Sanitizer.sanitizeForRichText(text.getValue()))
```
A possible extraction is:
```java
// in Sanitizer.java

public static Text sanitizeForRichText(Text text) {
    return ...;
}
```

### PR DESCRIPTION
**Pull request by mohamed-moanis**
Fixes #6211

### CODE REVIEW

**Comment by damithc** [comment]
@mohamed-moanis let us know when ready for review.

**Comment by mohamed-moanis** [comment]
It's Ready

File: src/main/java/teammates/common/datatransfer/FeedbackResponseCommentAttributes.java [hunk 1]
```diff
@@ -191,9 +191,7 @@ public String getJsonString() {
     @Override
     public void sanitizeForSaving() {
         this.commentText = Sanitizer.sanitizeTextField(this.commentText);
```
Comment (wkurniawan07) [inline src/main/java/teammates/common/datatransfer/FeedbackResponseCommentAttributes.java]: It seems like we already have the function for this purpose, after all. See if this function differs from the newly defined `sanitizeForRichText`, and if we can decommission one of them (i.e just use one function for all purposes).

File: src/main/java/teammates/common/util/Sanitizer.java [hunk 2]
```diff
@@ -142,6 +142,18 @@ public static String sanitizeForRichText(String content) {
         }
         return policy.sanitize(content);
     }
+    
+    /**
+     * Sanitizes the {@link com.google.appengine.api.datastore.Text} with rich-text.
+     * Removes disallowed elements based on defined policy.
+     * @return A new sanitized {@link com.google.appengine.api.datastore.Text} or null if the input was null.
```
Comment (wkurniawan07) [inline src/main/java/teammates/common/util/Sanitizer.java]: Is it necessary to use fully qualified name? Unless there are name collisions, using just the class name would work just fine.

Comment (mohamed-moanis) [inline src/main/java/teammates/common/util/Sanitizer.java]: I think this is more a question of preferences

File: src/main/java/teammates/common/datatransfer/FeedbackResponseCommentAttributes.java (hunk 1 above)
Comment (mohamed-moanis) [inline src/main/java/teammates/common/datatransfer/FeedbackResponseCommentAttributes.java]: `sanitizeTextField` removes leading and trailing white spaces only. While `sanitizeForRichText` uses html sensitization, so we can't decommission one of them. But for this case we can refactor it and add the call to `sanitizeTextField` within `sanitizeForRichText`.

Comment (mohamed-moanis) [inline src/main/java/teammates/common/datatransfer/FeedbackResponseCommentAttributes.java]: So `sanitizeForRichText` will do something like this

`return new Text(Sanitizer.sanitizeForRichText(sanitizeTextField(text.getValue())));`

**Comment by mohamed-moanis** [comment]
@wkurniawan07 please check now

File: src/main/java/teammates/common/datatransfer/CommentAttributes.java [hunk 3]
```diff
@@ -224,9 +222,7 @@ public void sanitizeForSaving() {
             recipients = sanitizedRecipients;
         }
         
-        if (commentText != null) {
-            this.commentText = new Text(Sanitizer.sanitizeForRichText(commentText.getValue()));
-        }
+        this.commentText = Sanitizer.sanitizeForRichText(commentText);
```
Comment (wkurniawan07) [inline src/main/java/teammates/common/datatransfer/CommentAttributes.java]: Line 213 (not shown on the diff) has
```java
this.commentText = Sanitizer.sanitizeTextField(this.commentText);
```
which should not be needed anymore?

**Comment by damithc** [comment]
@belyabl9 any input from you before we merge this?
@mohamed-moanis perhaps you can sync this branch in the meantime?

**Comment by belyabl9** [comment]
@damithc , changes looks good

**Comment by mohamed-moanis** [comment]
Ready for merging @wkurniawan07

**Pull request merged**